pkg/controller/common/password/params.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package password
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sethvargo/go-password/password"
+)
+
+// GeneratorParams defines the parameters for generating random passwords.
+type GeneratorParams struct {
+	LowerLetters string
+	UpperLetters string
+	Digits       string
+	Symbols      string
+	Length       int
+}
+
+func NewGeneratorParams(allowedCharacter string, maxLength int) (GeneratorParams, error) {
+	generatorParams, other := categorizeAllowedCharacters(allowedCharacter)
```

Comment:
It is not possible to define a password policy without lower case letters: 

`--password-allowed-characters=UVWXYZ0123456789`  will still lead to passwords with lower case letters. This is a limitation of the password generator library. 

This should be validated here.